source/backend/dal/Repositories/ResearchFileRepository.cs

```diff
@@ -138,6 +139,120 @@ public PimsResearchFile Update(PimsResearchFile researchFile)
             return researchFile;
         }
 
+        /// <summary>
+        /// Retrieves the research file with the specified id last update information.
+        /// </summary>
+        /// <param name="id"></param>
+        /// <returns></returns>
+        public LastUpdatedByModel GetLastUpdateBy(long id)
+        {
+            var lastUpdatedByAggregate = new List<LastUpdatedByModel>();
+            var fileLastUpdatedBy = this.Context.PimsResearchFiles.AsNoTracking()
+                .Where(r => r.ResearchFileId == id)
+                .Select(r => new LastUpdatedByModel()
+                {
+                    ParentId = id,
+                    AppLastUpdateUserid = r.AppLastUpdateUserid,
+                    AppLastUpdateUserGuid = r.AppLastUpdateUserGuid,
+                    AppLastUpdateTimestamp = r.AppLastUpdateTimestamp,
+                })
+                .OrderByDescending(lu => lu.AppLastUpdateTimestamp)
+                .Take(1)
+                .ToList();
+            lastUpdatedByAggregate.AddRange(fileLastUpdatedBy);
+
+            var documentsLastUpdatedBy = this.Context.PimsResearchFileDocuments.AsNoTracking()
+                .Where(rd => rd.ResearchFileId == id)
+                .Include(rd => rd.Document)
+                .Select(rd => new LastUpdatedByModel()
+                {
+                    ParentId = id,
+                    AppLastUpdateUserid = rd.Document.AppLastUpdateUserid,
+                    AppLastUpdateUserGuid = rd.Document.AppLastUpdateUserGuid,
+                    AppLastUpdateTimestamp = rd.Document.AppLastUpdateTimestamp,
+                })
+                .OrderByDescending(lu => lu.AppLastUpdateTimestamp)
+                .Take(1)
+                .ToList();
+            lastUpdatedByAggregate.AddRange(documentsLastUpdatedBy);
+
+            // This is needed to get the document last-updated-by from the document that where deleted
+            var documentsHistoryLastUpdatedBy = this.Context.PimsResearchFileDocumentHists.AsNoTracking()
+                .Where(rdh => rdh.ResearchFileId == id)
+                .Select(rdh => new LastUpdatedByModel()
+                {
+                    ParentId = id,
+                    AppLastUpdateUserid = rdh.AppLastUpdateUserid, // TODO: Update this once the DB tracks the user
```

Comment:
is this specific to the document history? only see this comment here.

Reply:
Is something that was missed during analysis. The deleted stamp is only retrieved from history but the triggers do not a user that deleted. This is something we might want everywhere.